tests/test_quantum_distance.py

```diff
@@ -109,56 +169,23 @@ def test_quantum_distance_equals_eucl_distance(self)-> None:
                 self.vectors_that_need_padding
             )
         ):
-            for j in range(self.n_sizes):
-                for k in range(args.n_samples -1):
-                    vector_1 = vectors_list[j][k]
-                    vector_2 = vectors_list[j][k + 1]
-                    vector_1 = normalise_vector(vector_1)
-                    vector_2 = normalise_vector(vector_2)
-                    if i == 1:
-                        vector_1 = pad_vector_with_zeros(vector_1)
-                        vector_2 = pad_vector_with_zeros(vector_2)
-                    quantum_distance = qd(
-                        vector_1,vector_2
-                    ).compute_quantum_distance()
-                    euclidean_distance = np.linalg.norm(vector_2 - vector_1)
-                    self.assertAlmostEqual(
-                        quantum_distance,
-                        euclidean_distance
-                    )
+            for k in range(self.n_samples -1):
+                vector_1 = vectors_list[k]
+                vector_2 = vectors_list[k + 1]
+                vector_1 = normalise_vector(vector_1)
+                vector_2 = normalise_vector(vector_2)
+                if i == 1:
+                    vector_1 = pad_vector_with_zeros(vector_1)
+                    vector_2 = pad_vector_with_zeros(vector_2)
+                quantum_distance = qd(
+                    vector_1,vector_2
+                ).compute_quantum_distance()
+                euclidean_distance = np.linalg.norm(vector_2 - vector_1)
+                self.assertAlmostEqual(
+                    quantum_distance,
+                    euclidean_distance
```

Comment:
This is a very long test. I feel you're doing 2 tests in 1, am i correct? The first with fixed vectors, the second not. I'd split this into 2 tests. You can say you test that specific vector pair in the first one and the other is a more general case.

Reply:
Also, as noted in another test, try to remove as many internal functions from this code as you can. Limit inter-dependencies!

Reply:
Split the test in two. 

For the internal functions see comment below.